docs/systemd.md

```diff
@@ -74,6 +74,10 @@ clustered mode and application preload.
 **Note:** Socket activation doesn't currently work on jruby. This is
 tracked in [#1367].
 
+**Note:** Do not run Puma with `bundle` when using with socket
+activation. This may cause issues with shell interactivity (Ref:
+[1499#issuecomment-481772886](https://github.com/puma/puma/issues/1499#issuecomment-481772886)
+
```

Comment:
So I wonder if the mentioned incompatibility with ruby 2.6 is really about the version of bundler it includes? 

As mentioned in the original issue https://github.com/puma/puma/issues/1499#issuecomment-358867492, does `bundle exec` with the `--keep-file-descriptors` flag also avoid the problem you were seeing? If so then you might change this to say something like: 

_Any wrapper scripts which `exec`, or other indirections in `ExecStart`, may result in activated socket file descriptors being closed before they reach the puma master process.  For example, if using `bundle exec`, pass the `--keep-file-descriptors` flag._

Reply:
Hi @dekellum

> So I wonder if the mentioned incompatibility with ruby 2.6 is really about the version of bundler it includes?

I don't know that. I'm using bundler 1.16.6 with ruby 2.5 and 1.17.2 with ruby 2.6

> As mentioned in the original issue #1499 (comment), does bundle exec with the --keep-file-descriptors flag also avoid the problem you were seeing? 

Yes, it does. Didn't try that before, I didn't even read that comment 🤦‍♂️ 

> Any wrapper scripts which exec, or other indirections in ExecStart, may result in activated socket file descriptors being closed before they reach the puma master process. For example, if using bundle exec, pass the --keep-file-descriptors flag.

Since keep-file-descriptors work, I will force-push in a moment